src/lib.rs

```diff
@@ -278,10 +294,43 @@ impl Transaction {
     }
 }
 
-impl_from_core_type!(Transaction, BitcoinTransaction);
-impl_from_ffi_type!(Transaction, BitcoinTransaction);
+impl_from_core_type!(Transaction, bitcoin::Transaction);
+impl_from_ffi_type!(Transaction, bitcoin::Transaction);
+
+#[derive(Clone, Default, uniffi::Enum)]
+#[non_exhaustive]
+pub enum Network {
```

Comment:
If we have to copy and paste enums from `rust-bitcoin` into here just to call `uniffi::Enum` I can see that getting annoying. Is it possible to use the UDL for direct exports of `rust-bitcoin` and nothing else?

Reply:
See my comment in the first commit

>"It is permitted to use this [`uniffi::Enum`] macro on a type that is
also defined in the UDL file as long as the two definitions are equal in
the names and ordering of variants and variant fields"

see: https://mozilla.github.io/uniffi-rs/latest/proc_macro/index.html#the-uniffienum-derive

I think it's possible to *only* use the UDL, but then it needs to manually be imported using [uniffi::use_udl_enum!(dependent_crate, EnumType);](https://mozilla.github.io/uniffi-rs/0.27/proc_macro/index.html#types-from-udl-based-dependent-crates) but repeating the pure enums (not the errors, for some reason) seems to be a better trade off

This is the only macro limitation that frustrates me. I prefer ProcMacros in every other circumstance and dealing with this trade off is worth it to me.